shared/src/test/diff/codegen/Polyfill.mls

```diff
@@ -0,0 +1,26 @@
+:js
+n = 123 with { x = 1 }
+//│ // Prelude
+//│ function withConstruct(target, fields) {
+//│   if (typeof target === "string" || typeof target === "number" || typeof target === "boolean" || typeof target === "bigint" || typeof target === "symbol") {
+//│     return Object.assign(target, fields);
+//│   }else {
+//│     const copy = Object.assign({}, target, fields);
+//│     Object.setPrototypeOf(copy, Object.getPrototypeOf(target));
+//│     return copy;
+//│   }
+//│ }
+//│ let res;
+//│ // Query 0
+//│ const n = withConstruct(123, {  x: 1  }); res = n;
+//│ // End of generated code
+//│ n: 123 & {x: 1}
+//│  = [Number: 123] { x: 1 }
```

Comment:
I wanted to forbid this in the type system eventually, but it turns out JavaScript can do it. Thankfully I had not done it yet! 🤪 

So, can we also extend things like numbers and booleans in classes?

Reply:
Yes, you can!

```js
class Test extends Number {
  constructor() {
    super();
    this.x = 0;
  }
}
// undefined
new Test()
// [Number (Test): 0] { x: 0 }
new Test() + 1
// 1
new Test() + 100
// 100
```